### PR TITLE
[WIP] Get database connection before backup

### PR DESCRIPTION
**Pull request by jerrywardlow**
https://github.com/Expensify/Expensify/issues/74850

Open to feedback, got some insight from Cole about this and got Travis tests passing, but am going to leave this as a [WIP] until plenty of folks have had their eyes on it and weighed in on if this is the best way to accomplish this goal.

Also stoked to be leveling up my C++ and getting a much better understanding of the inner workings of Bedrock, looking forward to more of this in the future.

### CODE REVIEW

**Comment by coleaeason** [comment]
Looks reasonable, have you tested?

File: main.cpp [hunk 1]
```diff
@@ -50,6 +50,14 @@ void VacuumDB(const string& db) { RetrySystem("sqlite3 " + db + " 'VACUUM;'"); }
 
 #define BACKUP_DIR "/var/tmp/"
 void BackupDB(const string& dbPath) {
+
+    // Initialize sqlite3 pointer '_db'
+    sqlite3* _db = NULL;
+
+    // Open database connection to prevent checkpointing
+    SINFO("Opening connection to " << dbPath);
+    SASSERT(sqlite3_open_v2(dbPath.c_str(), &_db, SQLITE_OPEN_READONLY, NULL));
```
Comment (coleaeason) [inline main.cpp]: We might want to pass `SQLITE_OPEN_NOMUTEX` here as well. I can't find a defined behavior anywhere for what happens if we have the connection still open in another thread and then open it here in singlethreaded mode. @tylerkaraszewski do you have any idea on this?

**Comment by tylerkaraszewski** [comment]
> We might want to pass SQLITE_OPEN_NOMUTEX

I'm not sure what this will (or won't) do. I agree we should test the code as-is and make sure it does what we think it does.

**Comment by coleaeason** [comment]
This isn't needed anymore

**Pull request closed without merging**